python/dgl/graphbolt/subgraph_sampler.py

```diff
@@ -34,9 +36,16 @@ def __init__(
         self,
         datapipe,
     ):
-        super().__init__(datapipe, self._sample)
+        super().__init__(datapipe, lambda x: x)
+
+        def _postprocess(minibatch):
+            delattr(minibatch, "seeds_timestamp")
+            return minibatch
 
-    def _sample(self, minibatch):
+        self.datapipe = self.datapipe.transform(_postprocess)
+        self.append_sampling_step(MiniBatchTransformer, self._preprocess)
```

Comment:
```suggestion
        self.datapipe = self.datapipe.transform(_preprocess)
        self.append_stages()
        self.datapipe = self.datapipe.transform(_postprocess)
```
```
def append_stages():
     self.datapipe = self.datapipe.transform(_sample)

def _sample(minibatch):
    (
        minibatch.input_nodes,
        minibatch.sampled_subgraphs,
    ) = self.sample_subgraphs(minibatch._seed_nodes, minibatch._seeds_timestamp)
```
In this way, we can keep existing api unchanged, and user can override either sample_subgraphs or append_stages.

Reply:
Sounds great!